parser/parser.go

```diff
@@ -6,127 +6,185 @@ import (
 	"errors"
 	"fmt"
 	"io"
-	"log/slog"
-	"slices"
+	"strings"
 )
 
 type Command struct {
 	Name string
 	Args string
+	bytes.Buffer
 }
 
-func (c *Command) Reset() {
-	c.Name = ""
-	c.Args = ""
-}
-
-func Parse(reader io.Reader) ([]Command, error) {
-	var commands []Command
-	var command, modelCommand Command
-
-	scanner := bufio.NewScanner(reader)
-	scanner.Buffer(make([]byte, 0, bufio.MaxScanTokenSize), bufio.MaxScanTokenSize)
-	scanner.Split(scanModelfile)
-	for scanner.Scan() {
-		line := scanner.Bytes()
-
-		fields := bytes.SplitN(line, []byte(" "), 2)
-		if len(fields) == 0 || len(fields[0]) == 0 {
-			continue
+func Parse(r io.Reader) ([]Command, error) {
+	var cmds []Command
+	var errs []error
+	parseCommands(r)(func(cmd Command, err error) bool {
```

Comment:
I personally think it would be easier to read this if parseCommands just returned an array of errors, rather than yielding.

Reply:
Fair. Once `rangefunc` is enabled, this will look identical to the current implemention but unless the caller of Parse is updated, there's no real benefit.